### PR TITLE
Find tables in using clause of delete statement

### PR DESCRIPTION
**Pull request by Tassosb**

### CODE REVIEW

File: spec/lib/parse_spec.rb [hunk 1]
```diff
@@ -1534,16 +1534,52 @@
...
+        UPDATE users SET name = users_new.name
+        FROM users_new
+        INNER JOIN join_table ON join_table.user_id = new_users.id
+        WHERE users.id = users_new.id
       SQL
       expect(query.warnings).to be_empty
-      expect(query.tables).to eq(['users', 'users_new'])
-      expect(query.select_tables).to eq(['users_new'])
+      expect(query.tables).to eq(['users', 'users_new', 'join_table'])
+      expect(query.select_tables).to eq(['users_new', 'join_table'])
```
Comment (Tassosb) [inline spec/lib/parse_spec.rb:1544]: This spec change is unrelated to the main change of the PR. I think the spec as written in this branch would have failed in a slightly older version of the code but now it is passing.

File: spec/lib/parse_spec.rb [hunk 2]
```diff
@@ -1534,16 +1534,52 @@
...
+      expect(query.warnings).to be_empty
+      expect(query.tables).to eq(['users', 'foo'])
+      expect(query.dml_tables).to eq(['users'])
+      expect(query.select_tables).to eq(['foo'])
+    end
+
+    it 'finds the table in the where subquery' do
+      query = described_class.parse(<<-SQL)
+        delete from users
+          where foo_id IN (select id from foo where action = 'delete');
```
Comment (olleolleolle) [inline spec/lib/parse_spec.rb:1574]: Guessing the style in the rest of the file is upper-case SQL:

```suggestion
        DELETE FROM users
          WHERE foo_id IN (SELECT id FROM foo WHERE action = 'delete');
```

File: spec/lib/parse_spec.rb [hunk 3]
```diff
@@ -1534,16 +1534,52 @@
...
+      SQL
+      expect(query.warnings).to be_empty
+      expect(query.tables).to eq(['users'])
+      expect(query.dml_tables).to eq(['users'])
+    end
+
+    it 'finds the used table' do
+      query = described_class.parse(<<-SQL)
+        delete from users using foo
+          where foo_id = foo.id and foo.action = 'delete';
```
Comment (olleolleolle) [inline spec/lib/parse_spec.rb:1563]: Style suggestion to harmonize:

```suggestion
        DELETE FROM users USING foo
          WHERE foo_id = foo.id AND foo.action = 'delete';
```

File: spec/lib/parse_spec.rb [hunk 4]
```diff
@@ -1534,16 +1534,52 @@
...
+      expect(query.select_tables).to eq(['users_new', 'join_table'])
       expect(query.dml_tables).to eq(['users'])
       expect(query.ddl_tables).to eq([])
     end
   end
 
+  describe 'parsing DELETE' do
+    it 'finds the deleted table' do
+      query = described_class.parse(<<-SQL)
+        delete from users;
```
Comment (olleolleolle) [inline spec/lib/parse_spec.rb:1553]:
```suggestion
        DELETE FROM users;
```

**Review by olleolleolle — COMMENTED** [review]
Would it work using upper-case SQL strings?

**Review by lfittl — APPROVED** [review]
